### PR TITLE
Must use mountlabel when creating builtin volumes

### PR DESCRIPTION
**Pull request by rhatdan**
Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

#### Does this PR introduce a user-facing change?

```release-note
If user modifies the filelabel security-opt, builtin volumes will now use the specified label.
```

### CODE REVIEW

**Comment by Luap99** [comment]
you need to rebase to fix unrelated CI errors (#17247)

File: version/version.go
Comment (Luap99) [inline version/version.go:30]: this must be removed

**Comment by rhatdan** [comment]
@containers/podman-maintainers PTAL

**Review by vrothberg — COMMENTED** [review]
LGTM

**Comment by rhatdan** [comment]
@Luap99 PTAL

File: pkg/domain/infra/abi/play.go
Comment (Luap99) [inline pkg/domain/infra/abi/play.go:249]: I don't think that can work correctly, here we loop over all content in the file.
I don't see any guarantee that a pod must be before the PersistentVolumeClaim. Right now it is just depends on the user, that seems wrong.

Why even change play kube? There is no test for it or any description why kube was changed? I understand the ` --security-opt` change but kube does not support it AFAIK.

File: pkg/domain/infra/abi/play.go
Comment (Luap99) [inline pkg/domain/infra/abi/play.go:289]: see other comment

**Review by Luap99 — CHANGES_REQUESTED** [review]

File: pkg/domain/infra/abi/play.go
Comment (rhatdan) [reply pkg/domain/infra/abi/play.go:249]: Kube yaml can support filetype labels (At least in Podman) Now.

I can change this to just pass in "", since it is arbitrary, but I am not sure that is correct in the general case.

**Comment by rhatdan** [comment]
@Luap99 PTAL

File: pkg/domain/infra/abi/play.go
Comment (Luap99) [inline pkg/domain/infra/abi/play.go:251]: I don't understand this, looking more into how this works we first sort the kinds in `sortKubeKinds()` so that pods always come last. This means that the `PersistentVolumeClaim` case is always executed before you get here.
Thus the mountlabel will always be empty.

As I read the code you handle it already correctly in `playKubePod()`, so I suggest you remove this part here and from `playKubePVC()`.

**Comment by vrothberg** [comment]
Friendly ping

**Comment by rhatdan** [comment]
@vrothberg @Luap99 @mheon PTAL
@containers/podman-maintainers PTAL

**Comment by TomSweeneyRedHat** [comment]
Happy greeen test buttons.  A quick scan and the code LGTM.  But I definitely want a head nod from @Luap99 before merging to make sure his concerns were answered.

File: pkg/domain/infra/abi/play.go
Comment (Luap99) [reply pkg/domain/infra/abi/play.go:252]: This comment is still open, please remove this unneeded code or explain what it does.

**Comment by rhatdan** [comment]
@Luap99 I would really like to get this merged, and can look at fixing the order of a kubernetes yaml definition to make sure the label is processed before the volumes are created.

**Comment by Luap99** [comment]
I don't want code that does nothing but confuses readers to be merged, please address my comment.

File: pkg/domain/infra/abi/play.go
Comment (rhatdan) [reply pkg/domain/infra/abi/play.go:252]: If there is a specified pod and it has a mount label it will be used. This will allow Kubernetes.YAML to specify a file label and have any builtin volumes match the specified label.

Comment (Luap99) [reply pkg/domain/infra/abi/play.go:252]: As described above the sorting makes sure that this never happens and AFAICT the sorting has to happen because volumes must be created before containers otherwise a container could not use them.

Comment (rhatdan) [reply pkg/domain/infra/abi/play.go:252]: Well we have to find a way to fix it by parsing the pod spec first before creating the volume, so it can take into effect the specified mountLabel.

Comment (rhatdan) [reply pkg/domain/infra/abi/play.go:252]: I want to add the ability to specify the mountLabel for a volume in quadlet, to handle .Volume files, perhaps we can do something there.  Might need to add flag `podman volume create --mount-type`

Comment (rhatdan) [reply pkg/domain/infra/abi/play.go:252]: Or specify it as an podman volume create opt mountType=qm_container_file_t ...

**Comment by rhatdan** [comment]
@Luap99 I removed the code you were complaining about.

**Review by Luap99 — APPROVED** [review]
/lgtm
/hold

**Comment by mheon** [comment]
Lint's angry about an extra trailing space

**Comment by mheon** [comment]
Is this safe for volumes being used by more than one container, given that it looks like we're copying the MCS part of the label, too?

**Comment by rhatdan** [comment]
As long as the containers are using the same file type, no way to truly deal with containers with different file types.  Unless we add a new type.

I have a goal of adding an option to create a volume with a specific file type.

This would allow us to have a third label shared between two container_file_t types.

Most likely this would not happen often, really only needed for RHIVOS.

**Pull request merged**
